### ISSUE TITLE
checkbox not getting checked/unchecked when enter is pressed, even with passing onKeyDown

### ISSUE THREAD

**Issue by abigaelonchiri-msft** [microsoft/fluentui#12517]
### Environment Information

- **Package version(s)**: (fill this out)
- **Browser and OS versions**: (fill this out)

### Please provide a reproduction of the issue in a codepen:

https://codepen.io/abbyonch/pen/vYOMPWB

#### Actual behavior:

When pressing enter checkbox should be checked or unchecked. When passing keyDown method , method is not getting called and checkbox is not checked. Also keyDown method doesnt taked checked parameter

#### Expected behavior:

Checkbox should be checked/ unchecked on enter

### Documentation describing expected behavior

Checkbox should be checked/ unchecked on enter

**Comment by aneeshack4** [comment]
@abigaelonchiri-msft This is by design - SPACE not ENTER should be used to check and uncheck. This is explained in #8608 particularly this [comment](https://github.com/microsoft/fluentui/issues/8608#issuecomment-480032258).
